app/javascript/pages/components/Browser.jsx

```diff
@@ -83,18 +89,36 @@ class Browser extends React.Component {
         </div>
         <div className="category-lists">
           <div className="container tight">
-            <DataMenu items={items} datasets={this.props.datasets} onMenuClick={this.handleMenuSelectedItem('menuOneSelectedItem')} />
+            <DataMenu
+              items={items}
+              datasets={this.props.datasets}
+              onMenuClick={(event) =>
+                this.handleMenuSelectedItem('menuOneSelectedItem', event)
+              }
+            />
             {items.length > 0 && menuOneSelectedItem ? (
               <DataMenu
                 datasets={this.props.datasets}
-                items={items.filter((item) => item.menuTitle === menuOneSelectedItem)[0].items}
-                onMenuClick={this.handleMenuSelectedItem('menuTwoSelectedItem')}
+                items={
+                  items.filter(
+                    (item) => item.menuTitle === menuOneSelectedItem
+                  )[0].items
+                }
+                onMenuClick={(event) =>
+                  this.handleMenuSelectedItem('menuTwoSelectedItem', event)
+                }
               />
             ) : null}
             {items.length > 0 && menuTwoSelectedItem ? (
               <DataMenu
-                items={items.filter((item) => item.menuTitle === menuOneSelectedItem)[0].items.filter((item) => item.menuTitle === menuTwoSelectedItem)[0].items}
-                onDatasetClick={this.handleDatasetClick()}
+                items={
+                  items
+                    .filter((item) => item.menuTitle === menuOneSelectedItem)[0]
+                    .items.filter(
+                      (item) => item.menuTitle === menuTwoSelectedItem
```

Comment:
This filter on `items` throws an error when clicking back to a first-menu topic while already viewing items on the third menu (if you only have the first two menus open, you can click to another highest-level topic with no problem)

Reply:
I actually had this fix for this in the old code. I re-introduced it and It seems to be working for me now but give it a shot.